uni/lib/view/exams/widgets/exam_filter_form.dart

```diff
@@ -6,14 +8,21 @@ import 'package:uni/model/providers/lazy/exam_provider.dart';
 
 class ExamFilterForm extends StatefulWidget {
   const ExamFilterForm(this.filteredExamsTypes, {super.key});
-
   final Map<String, bool> filteredExamsTypes;
 
   @override
   ExamFilterFormState createState() => ExamFilterFormState();
 }
 
 class ExamFilterFormState extends State<ExamFilterForm> {
+  void _changeFilteredExamList(String key, {bool? value}) {
+    setState(() {
+      widget.filteredExamsTypes[key] = value!;
+      Provider.of<ExamProvider>(context, listen: false)
+          .setFilteredExams(widget.filteredExamsTypes);
+    });
```

Comment:
You seem to still be modifying the widget prop. Perhaps I could be of help sometime in the next few days so that you get what I was saying?

Reply:
Okay.